packages/mongo-nav/src/project-nav/ProjectNav.spec.tsx

```diff
@@ -66,29 +67,48 @@ describe('packages/mongo-nav/src/project-nav', () => {
 
   let onProjectChange: jest.Mock;
 
+  let fetchMock: jest.Mock;
+  let originalFetch: (
+    input: RequestInfo,
+    init?: RequestInit | undefined,
+  ) => Promise<Response>;
+
+  let fetchData: Array<object>;
+
   beforeEach(() => {
     onProjectChange = jest.fn();
+    fetchMock = jest.fn();
+    fetchData = [];
+    fetchMock.mockResolvedValue({
+      ok: true,
+      json: () => Promise.resolve(fetchData),
+    });
+    originalFetch = window.fetch;
+    window.fetch = fetchMock;
   });
 
   afterEach(() => {
+    window.fetch = originalFetch;
     jest.restoreAllMocks();
     cleanup();
   });
 
-  const renderComponent = (props = {}) => {
-    setQueries(
-      render(
-        <ProjectNav
-          data={projects}
-          constructProjectURL={constructProjectURL}
-          urls={urlDefaults}
-          activeProduct="cloud"
-          hosts={hostDefaults}
-          onProjectChange={onProjectChange}
-          {...props}
-        />,
-      ),
-    );
+  const renderComponent = async (props = {}) => {
+    await act(async () => {
```

Comment:
Not sure, but is there a reason we need the callback to act to be an async function? I'm not sure what we get for doing that.

Reply:
Without the inner async we get this error:
![image](https://user-images.githubusercontent.com/322576/77686300-19068000-6f73-11ea-9b32-d0863e2c851a.png)

Without the outer await we get this:

![image](https://user-images.githubusercontent.com/322576/77686318-24f24200-6f73-11ea-8f16-fe65cb0f03f7.png)


It's because we're waiting for a Promise to resolve inside the component.

Reply:
Wow that's unintuitive. Okay 😬lol